### PR TITLE
[seal] Updates port to 3.5.8

### PR DESCRIPTION
**Pull request by pjattke**
Updates SEAL port to version 3.5.8.

### CODE REVIEW

File: ports/seal/portfile.cmake [hunk 1]
```diff
@@ -7,21 +7,22 @@ vcpkg_fail_port_install(ON_TARGET "uwp")
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" SEAL_BUILD_STATIC)
```
Comment (NancyLi1013) [inline ports/seal/portfile.cmake:7]:
```
set(SEAL_VERSION_MINOR 4)
set(SEAL_VERSION_MICRO 5)
```
The version also needs to be updated.

File: ports/seal/portfile.cmake [hunk 2]
```diff
@@ -7,21 +7,22 @@ vcpkg_fail_port_install(ON_TARGET "uwp")
...
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" SEAL_BUILD_STATIC)
```
Comment (NancyLi1013) [inline ports/seal/portfile.cmake:8]: It seems that `SEAL_BUILD_STATIC` doesn't exist on this version. Could you please check this again?

File: ports/seal/portfile.cmake [hunk 3]
```diff
@@ -30,11 +31,11 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
 )
 
 vcpkg_configure_cmake(
-    SOURCE_PATH ${SOURCE_PATH}/native/src
+    SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA
     OPTIONS ${FEATURE_OPTIONS}
         -DALLOW_COMMAND_LINE_BUILD=ON
-        -DSEAL_LIB_BUILD_TYPE=${SEAL_LIB_BUILD_TYPE}
+        -DBUILD_SHARED_LIBS=${BUILD_SHARED_LIBS}
```
Comment (NancyLi1013) [inline ports/seal/portfile.cmake:38]: This seems to be not needed. 
```suggestion
```
Please refer to this doc https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#choose-either-static-or-shared-binaries

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/seal/CONTROL [hunk 4]
```diff
@@ -1,9 +1,9 @@
 Source: seal
-Version: 3.4.5-1
+Version: 3.5.8
 Homepage: https://github.com/microsoft/SEAL
 Description: Microsoft SEAL is an easy-to-use and powerful homomorphic encryption library.
 Default-Features: zlib
```
Comment (NancyLi1013) [inline ports/seal/CONTROL:5]:
```suggestion
Supports: !uwp
Default-Features: zlib
```
Seems this port currently doesn't support uwp, So we can add `Supports `filed here. Please also remove 
```
seal:arm-uwp=fail
seal:x64-uwp=fail
```
from ci.baseline.txt.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @pjattke 

Could you please resolve the conflicts and also address the review suggestions?

**Comment by pjattke** [comment]
@NancyLi1013 Sorry, I forgot about that. Has in the meantime already be updated to the latest version (3.5.9) in #13598.

**Comment by NancyLi1013** [comment]
Thanks for your feedback and also for your kindly reminder. I didn't noticed this point yet.

**Pull request closed without merging**